app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :atransactions, foreign_key: :author_id, dependent: :destroy
+  has_many :categories, foreign_key: :author_id, dependent: :destroy
+
+  validates :name, presence: true, length: { minimum: 2, maximum: 25 }
```

Comment:
- [OPTIONAL] It will be better to validate all the user input (password, confirm_pass ...) to make sure that anyone with bad intentions cannot compromise their app.